substrate/client/transaction-pool/src/fork_aware_txpool/tx_mem_pool.rs

```diff
@@ -257,15 +344,21 @@ where
 		metrics: PrometheusMetrics,
 		max_transactions_count: usize,
 		max_transactions_total_bytes: usize,
-	) -> Self {
-		Self {
-			api,
-			listener,
-			transactions: Default::default(),
-			metrics,
-			max_transactions_count,
-			max_transactions_total_bytes,
-		}
+	) -> (Self, TxMemPoolBlockingTask) {
+		let (sync_channel, rx) = sync_bridge_channel();
```

Comment:
Not super convinced that we need this. We should be able to use `block_on`. But I am not passionate enough about this to insist :D. So do as you like.

Reply:
I had iteration using: [blocking_send](https://docs.rs/tokio/latest/src/tokio/sync/mpsc/bounded.rs.html#1037). And yes, it crashed. Don't really like what I did here, but don't see a better way to solve it.

We could change all fn to be async but it would require some refactoring around each use case.

Reply:
https://github.com/paritytech/polkadot-sdk/issues/8912